scripts/commands/brew-sync.ts

```diff
@@ -0,0 +1,80 @@
+import { exec, logFail, logInfo, logSuccess, readJsonFile, writeJsonFile } from '../functions/index';
+import { Command, CommandConfig, EnvironmentConfig } from '../models';
+
+export const brewSync: Command = async (config) => {
+  logInfo('BrewSync process initiated...');
+  try {
+    const data = await readJsonFile(config.configPath) as EnvironmentConfig;
+    const { casks, formulae } = await getUpdatedBrewItems(data) ?? { casks: [], formulae: [] };
+
+    // TODO - Better way to handle this?
+    if (!casks.length && !formulae.length) {
+      return;
+    }
+    if (casks.length) {
+      data.commands.forEach((command: CommandConfig) => {
+        if (command.key === 'brew-install' && command.value.args === '--cask') {
+          command.value.items = casks;
+        }
+      });
+    }
+    if (formulae.length) {
+      data.commands.forEach((command: CommandConfig) => {
+        if (command.key === 'brew-install' && !command.value.args) {
+          command.value.items = formulae;
+        }
+      });
+    }
+
+    logInfo('New items detected, Syncronization in 3... 2... 1... 🚀');
+    await writeJsonFile(config.configPath, data)
+      .then(() => logSuccess('Brew items syncronized!'));
+  }
+  catch (error) {
+    logFail(`Error syncing brew items: ${error}`);
+  }
+}
+
+
+const getUpdatedBrewItems = async (data: EnvironmentConfig) => {
```

Comment:
I had these all in a single methods originally and I broke them up because I thought you would tell me the exact opposite since we had just discussed SOLID principles 😅

Reply:
That is fair. I would argue the thing it does is add missing brew items, but I will admit that "single responsibility" can definitely be up for interpretation.

Personally I like SOLID and design patterns, but [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) is my one true religion.